File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    for (let i = 1; i <= yellow**0.5; i++){
        if (yellow % i == 0){
            let w = (yellow / i) + 2
            let h = i + 2
            
            if (brown == ((w + h - 2) * 2)) return [w, h]
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    
    for v in range(3, int(total**0.5)+1):
        if total % v == 0:
            h = int(total / v)
            if brown == (h + v - 2) * 2: return [h,v]
        
    return []